test/15_target-hostapp.ts

```diff
@@ -340,73 +342,87 @@ export default () => {
-				// if a device is preprovisioned and pinned to a release
-				// less than the version it initially checks in with, make sure the downgrade doesn't persist
-				const { body } = await supertest(admin)
-					.get(
-						`/${version}/device(${preprovisionedDevice.id})?$select=should_be_operated_by__release`,
-					)
-					.expect(200);
-				expect(body.d[0]).to.have.property(
-					'should_be_operated_by__release',
-					null,
-				);
+				it(`should remove the target preprovisioned hostapp and link with the reported upgraded ${titlePart} release, if it is an implied downgrade`, async () => {
+					// if a device is preprovisioned and pinned to a release
+					// less than the version it initially checks in with, make sure the downgrade doesn't persist
+					const preprovisionedDevice = await fakeDevice.provisionDevice(
+						admin,
+						applicationId,
+					);
+					await supertest(admin)
+						.patch(`/${version}/device(${preprovisionedDevice.id})`)
+						.send({ should_be_operated_by__release: prodNucHostappReleaseId })
+						.expect(200);
 
-				await supertest(admin)
-					.patch(`/${version}/device(${preprovisionedDevice.id})`)
-					.send({ should_be_operated_by__release: prodNucHostappReleaseId })
-					.expect(200);
-				const devicePatchBody = {
-					local: {
+					// run the actual test
+					const devicePatchBody = {
 						// this version is greater than the one provided by prodNucHostappReleaseId
-						os_version: 'balenaOS 2.50.1+rev1',
-						os_variant: 'prod',
-					},
-				};
-				await preprovisionedDevice.patchStateV2(devicePatchBody);
-				const body2 = await supertest(admin)
-					.get(
-						`/${version}/device(${preprovisionedDevice.id})?$select=should_be_operated_by__release`,
-					)
-					.expect(200);
-				expect(body2.body.d[0]['should_be_operated_by__release'].__id).to.equal(
-					upgradeReleaseId,
-				);
+						local: {
+							os_version: osVersion,
+							os_variant: 'prod',
+						},
+					};
+					await preprovisionedDevice.patchStateV2(devicePatchBody);
+					await expectResourceToMatch(
+						pineUser,
+						'device',
+						preprovisionedDevice.id,
+						{
+							should_be_operated_by__release: { __id: getUpgradeReleaseId() },
+						},
+					);
+				});
 			});
 
 			it('should succeed in PATCHing device to ESR release', async () => {
 				await supertest(admin)
-					.patch(`/${version}/device(${device.id})`)
-					.send({ should_be_operated_by__release: esrHostappReleaseId })
+					.patch(`/${version}/device(${device1.id})`)
+					.send({ should_be_operated_by__release: esrTagOnlyHostAppReleaseId })
 					.expect(200);
 				const { body } = await supertest(admin).get(
-					`/${version}/device(${device.id})?$select=should_be_operated_by__release`,
+					`/${version}/device(${device1.id})?$select=should_be_operated_by__release`,
 				);
 				expect(body.d[0]).to.not.be.undefined;
 				expect(body.d[0]['should_be_operated_by__release'].__id).to.equal(
```

Comment:
Can you also use `expectResourceToMatch()` here?

Reply:
Yes :)
I was just trying to keep the diff as short as possible to be kind w/ the reviewers, but since you asked ..... :) :tada: